File: 백준/Silver/10845. 큐/큐.py
import sys
from queue import Queue

n = int(input())

que = Queue()

for i in range(n) :
  data = sys.stdin.readline().split()
  if(data[0] == "push") :
    que.put(data[1])
  elif(data[0] == "pop") :
    if que.qsize() < 1 :
      print(-1)
    else :
      print(que.get())
  elif(data[0] == "size") :
    print(que.qsize())
  elif(data[0] == "empty") :
    if que.qsize() < 1 :
      print(1)
    else :
      print(0)
  elif(data[0] == "front") :
    if que.qsize() < 1 :
      print(-1)
    else :
      print(que.queue[0])
  elif(data[0] == "back") :
    if que.qsize() < 1 :
      print(-1)
    else :
      print(que.queue[-1])

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
#include <string>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    queue<int> Q;
    vector<int> V;

    for (int i = 0; i < n; i++){
        string str;
        cin >> str;
        if (str == "push") {
            int j;
            cin >> j;
            Q.push(j);
        }
        else if (str == "pop") {
            if (Q.empty()) V.push_back(-1);
            else {
                V.push_back(Q.front());
                Q.pop();
            }
        }
        else if (str == "size") {
            V.push_back(Q.size());
        }
        else if (str == "empty") {
            V.push_back(Q.empty());
        }
        else if (str == "front") {
            if (Q.empty()) V.push_back(-1);
            else V.push_back(Q.front());
        }
        else {
            if (Q.empty()) V.push_back(-1);
            else V.push_back(Q.back());
        }
    }

    for (int i = 0; i < V.size(); i++) {
        cout << V[i] << "\n";
    }

    return 0;
}
